### PR TITLE
Feat: styles

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

File: src/core/component/q-component-ctx.ts [hunk 1]
```diff
@@ -101,3 +108,20 @@ export function getHostElement(element: Element): HTMLElement | null {
...
+  if (style) {
+    const styleId = styleKey(style as any)!;
+    const document = ctx.hostElement.ownerDocument;
+    const head = document.querySelector('head')!;
+    if (!head.querySelector(`style[q\\:style="${styleId}"]`)) {
+      const styleImport = Promise.resolve(qImport<string>(document, style));
+      styleImport.then((styles: string) => {
+        const style = document.createElement('style');
+        style.setAttribute('q:style', styleId);
+        style.textContent = styles.replace(/�/g, styleId);
```
Comment (adamdbradley) [inline src/core/component/q-component-ctx.ts:122]: What is `/�/g`?

File: src/core/render/q-render.unit.tsx [hunk 2]
```diff
@@ -304,14 +304,18 @@ describe('q-render', () => {
         expectRendered(
           <hello-world
             on:q-render={HelloWorld.onRender}
-            q:style={HelloWorld.styles as any}
+            q:sstyle={HelloWorld_styles}
```
Comment (adamdbradley) [inline src/core/render/q-render.unit.tsx:307]: is `q:sstyle` correct?

File: src/core/util/markers.ts [hunk 3]
```diff
@@ -40,7 +40,12 @@ export const enum AttributeMarker {
   /**
    * Component Styles.
    */
-  ComponentStyles = 'q:style',
+  ComponentStyles = 'q:sstyle',
```
Comment (adamdbradley) [inline src/core/util/markers.ts:43]: Ok guess it's correct, but why the change from `q:style` to `q:sstyle`?

**Review by adamdbradley — APPROVED** [review]

File: src/core/render/q-render.unit.tsx (hunk 2 above)
Comment (mhevery) [reply src/core/render/q-render.unit.tsx:307]: Scoped Style. Not loving it, but for now until we figure out something better.

File: src/core/util/markers.ts (hunk 3 above)
Comment (mhevery) [reply src/core/util/markers.ts:43]: So we can have both scoped and unscoped styles.
